### ISSUE TITLE
`for` is working for `cmds` but not for `deps`

### ISSUE THREAD

**Issue by leobenkel** [go-task/task#1549]
- Task version:
```
task --version
Task version: v3.35.1 (h1:zjQ3tLv+LIStDDTzOQx8F97NE/8FSTanjZuwgy/hwro=)
```
- Operating system:
```
uname -a
Linux xxx 6.5.0-25-generic #25~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Feb 20 16:09:15 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
```

so when i do:

```
  build_module:
    requires:
      vars:
      - MODULE_NAME
      - ROOT_DIR
    vars:
      MODULE_NAME: '{{ .MODULE_NAME | default "" }}'
    preconditions:
      - sh: test -f {{.ROOT_DIR}}/modules/{{.MODULE_NAME}}/build.sh
    cmds:
      - echo "Building the {{.MODULE_NAME}} image"
      - echo "cd {{.ROOT_DIR}}/modules/{{.MODULE_NAME}}"

  build_all:
    vars:
      MODULES: a b c
    cmds:
      - for: [a, b, c]
        task: build_module
        vars:
          MODULE_NAME: '{{.ITEM}}'
```

it works 

but if you replace `cmds` by `deps` it does not work anymore.

and i have tried with `for: {var: MODULES}` and it behave the same as well: works with `cmds` but not with `deps`

**Comment by pd93** [comment]
Hi @leobenkel. Unfortunately, our docs are not currently versioned so this has been added to the site before it has been released. The looping over `deps` functionality was only merged to `main` a few days ago (#1541) and will be released in the next version of Task.

Also in the next release, we will be changing how we version our docs so that this kind of confusion doesn't keep happening.

**Comment by leobenkel** [comment]
ah ok, thank you !
